docs/generic-skills.md

```diff
@@ -0,0 +1,275 @@
+
+Moreover, this example provides a way to write your code to collect data if you want. 
+You can modify the `has_sensor` variable in `skill.yaml` file of the generic_seller skill to True. Then you have to implement the method `collect_data(self)` inside the strategy.py file. 
+### Launch an OEF node
+In a separate terminal, launch a local OEF node (for search and discovery).
+``` bash
+python scripts/oef/launch.py -c ./scripts/oef/launch_config.json
+```
+
+Keep it running for all the following demos.
+
+## Demo 1: Fetch.ai ledger payment
+
+A demo to run a scenario with a true ledger transaction on Fetch.ai `testnet`. This demo assumes the buyer
+trusts the seller AEA to send the data upon successful payment.
+
+### Create the seller AEA (ledger version)
+
+Create the AEA that will provide data.
+
+``` bash
+aea create my_seller_aea
+cd my_seller_aea
+aea add connection fetchai/oef:0.1.0
+aea add skill fetchai/generic_seller:0.1.0
+aea install
+```
+
+### Create the buyer client (ledger version)
+
+In another terminal, create the AEA that will query the seller AEA.
+
+``` bash
+aea create my_buyer_aea
+cd my_buyer_aea
+aea add connection fetchai/oef:0.1.0
+aea add skill fetchai/generic_buyer:0.1.0
+aea install
+```
+
+Additionally, create the private key for the buyer AEA.
+```bash
+aea generate-key fetchai
+aea add-key fetchai fet_private_key.txt
+```
+
+### Update the AEA configs
+
+Both in `my_seller_aea/aea-config.yaml` and
+`my_buyer_aea/aea-config.yaml`, replace `ledger_apis: {}` with the following.
+
+``` yaml
+ledger_apis:
+  fetchai:
+    network: testnet
+```
+
+### Update the seller AEA skill configs
+
+In `my_seller_aea/vendor/fetchai/generi_seller/skill.yaml`, replace the `data` with your data:
+```yaml
+shared_classes:
+  strategy:
+    class_name: Strategy
+    args:
+      total_price: 10
+      seller_tx_fee: 0
+      currency_id: 'FET'
+      ledger_id: 'fetchai'
+      is_ledger_tx: True
+      has_sensor: False
+      data:
+        foo: {'bar': 'this is test'}
+        test: 20
+```
+
+### Fund the buyer AEA
+
+Create some wealth for your buyer AEA on the Fetch.ai `testnet`. (It takes a while).
+``` bash
+aea generate-wealth fetchai
+```
+
+## Run the AEAs
+
+You can change the end point's address and port by modifying the connection's yaml file (my_thermometer_aea/connection/oef/connection.yaml)
+
+Under config locate :
+
+```bash
+addr: ${OEF_ADDR: 127.0.0.1}
+```
+ and replace it with your ip (The ip of the machine that runs the oef image.)
+
+Run both AEAs from their respective terminals
+
+```bash 
+aea add connection fetchai/oef:0.1.0
+aea install
+aea run --connections fetchai/oef:0.1.0
+```
+You will see that the AEAs negotiate and then transact using the Fetch.ai testnet.
+
+## Delete the AEAs
+When you're done, go up a level and delete the AEAs.
+```bash 
+cd ..
+aea delete my_seller_aea
+aea delete my_buyer_aea
+```
+## Demo instructions 2: Ethereum ledger payment
```

Comment:
Rather than repeating the guide twice, we should in the ledger specific section have either/or for fetchai/ethereum, otherwise there is too much duplication. Alternatively, in this section just point out what would need to be changed above (might make it even clearer)